.github/workflows/notebook-test.yml

```diff
@@ -12,15 +12,32 @@
 
 name: Test notebooks
 on:
+  # Three event types can trigger this workflow:
+  #  1. Pull request (opened, synchronize)
+  #     On creating or pushing to a PR branch, this action runs any changed
+  #     notebooks and reports pass / fail. This is to catch broken code
+  #     examples.
+  #  2. Pull request (labeled)
+  #     On adding the `🤖 Auto execute` label (id: 6409509633) to a PR, this
+  #     workflow executes any changed notebooks, then pushes the updates to the
+  #     PR branch. This makes it easy to quickly get notebooks executed in the
+  #     CI environment.
+  #  3. Workflow dispatch
+  #     When triggered by a workflow_dispatch event, we execute all notebooks,
+  #     and make a PR to the branch with the updated notebooks. This is handy
+  #     when upgrading qiskit package versions.
   pull_request:
     paths:
       - "docs/**/*.ipynb"
       - "!docs/api/**/*"
+    types: [opened, synchronize, labeled]
+  issue_comment:
   workflow_dispatch:
 jobs:
   execute:
     name: Execute notebooks
     runs-on: ubuntu-latest
+    if: github.event.action != 'labeled' || github.event.label.id == 6409509633
```

Comment:
I chose to go by label ID so we can change the name if needed. Note if you delete the label this feature will stop working.